src/test/java/teammates/storage/api/NotificationsDbTest.java

```diff
@@ -0,0 +1,272 @@
+package teammates.storage.api;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.test.BaseTestCaseWithLocalDatabaseAccess;
+
+/**
+ * SUT: {@link NotificationsDb}.
+ */
+public class NotificationsDbTest extends BaseTestCaseWithLocalDatabaseAccess {
+
+    private final NotificationsDb notificationsDb = NotificationsDb.inst();
+    private final Map<String, NotificationAttributes> typicalNotifications = getTypicalDataBundle().notifications;
+
+    // Notification attributes will be assigned with value at setup()
+    private NotificationAttributes n;
+
+    @BeforeMethod
+    public void setup() throws Exception {
+        for (NotificationAttributes n : typicalNotifications.values()) {
+            notificationsDb.createEntity(n);
+        }
+        n = typicalNotifications.get("notification1");
+    }
```

Comment:
While `notification1` is often used, I feel it is better to just declare this locally inside the test method itself each time it's used.

Reply:
This change is based on one of your previous comments that initialization should be done at `BeforeMethod`. Anyway it has been rolled back for this.

Reply:
Ah I see, sorry for not being clear.